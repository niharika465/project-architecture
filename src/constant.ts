// to add any global variables

export const IMAGE_WIDTH = 250;
export const IMAGE_HEIGHT = 200;
