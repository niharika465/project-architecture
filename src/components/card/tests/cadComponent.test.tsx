import { render, screen, fireEvent } from '@testing-library/react';
import CardComponent from '../cardComponent';
import { IMAGE_HEIGHT, IMAGE_WIDTH } from '../../../constant';

const mockProduct = {
  id: 1,
  title: 'Test Product',
  image: 'https://example.com/image.jpg',
  price: 10,
  description: 'Test product description',
  category: 'Mens',
  rating: { rate: 4, count: 10 },
};

const mockHandleClick = jest.fn();

describe('CardComponent', () => {
  beforeEach(() => {
    render(
      <CardComponent product={mockProduct} handleClick={mockHandleClick} />
    );
  });

  it('renders the product title', () => {
    const titleElement = screen.getByText(/Test Product/i);
    expect(titleElement).toBeInTheDocument();
  });

  it('matches the snapshot', () => {
    const { asFragment } = render(
      <CardComponent product={mockProduct} handleClick={mockHandleClick} />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders the product image', () => {
    const imageElement = screen.getByRole('img', { name: /Test Product/i });
    expect(imageElement).toHaveAttribute('src', mockProduct.image);
    expect(imageElement).toHaveAttribute('height', IMAGE_HEIGHT);
    expect(imageElement).toHaveAttribute('width', IMAGE_WIDTH);
  });

  it('calls handleClick with the correct id when clicked', () => {
    const cardElement = screen
      .getByText(/Test Product/i)
      .closest('div') as HTMLDivElement;
    fireEvent.click(cardElement);
    expect(mockHandleClick).toHaveBeenCalledWith(mockProduct.id);
  });
});
