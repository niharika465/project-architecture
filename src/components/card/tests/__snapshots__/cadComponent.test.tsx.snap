// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardComponent matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <h2
      class="card__name"
    >
      Test Product
    </h2>
    <img
      alt="Test Product"
      height="200"
      src="https://example.com/image.jpg"
      width="250"
    />
  </div>
</DocumentFragment>
`;
